src/managers/organization/authentication-by-peers.ts

```diff
@@ -1,314 +0,0 @@
-  if (otherInternalUsers.length > 0) {
-    const user_label =
-      !given_name && !family_name ? email : `${given_name} ${family_name}`;
-    const otherInternalUsersSample = sampleSize(
-      otherInternalUsers,
-      NOTIFY_ALL_MEMBER_LIMIT,
-    );
-    await sendMail({
-      to: otherInternalUsersSample.map(({ email }) => email),
-      subject: "Votre organisation sur MonComptePro",
-      template: "join-organization",
-      params: {
-        user_label,
-        libelle: cached_libelle,
-        email,
-        is_external: is_external,
-      },
-      senderEmail: "notifications@moncomptepro.beta.gouv.fr",
-    });
-
-    authentication_by_peers_type = "all_members_notified";
-  } else {
-    authentication_by_peers_type = "is_the_only_active_member";
-  }
-
-  return await updateUserOrganizationLink(organization_id, user_id, {
-    authentication_by_peers_type,
-  });
-};
-
-export const markAsWhitelisted = async ({
-  user_id,
-  organization_id,
-}: {
-  user_id: number;
-  organization_id: number;
-}) => {
-  const organizationUsers = await getUsers(organization_id);
-  const user = organizationUsers.find(({ id }) => id === user_id);
-  const organization = await findOrganizationById(organization_id);
-
-  // The user should be in the organization already
-  if (isEmpty(user) || isEmpty(organization)) {
-    throw new NotFoundError();
-  }
-
-  return await updateUserOrganizationLink(organization_id, user_id, {
-    authentication_by_peers_type: "deactivated_by_whitelist",
-  });
-};
-
-export const markAsGouvFrDomain = async ({
-  user_id,
-  organization_id,
-}: {
-  user_id: number;
-  organization_id: number;
-}) => {
-  const organizationUsers = await getUsers(organization_id);
-  const user = organizationUsers.find(({ id }) => id === user_id);
-  const organization = await findOrganizationById(organization_id);
-
-  // The user should be in the organization already
-  if (isEmpty(user) || isEmpty(organization)) {
-    throw new NotFoundError();
-  }
-
-  return await updateUserOrganizationLink(organization_id, user_id, {
-    authentication_by_peers_type: "deactivated_by_gouv_fr_domain",
-  });
-};
-
-export const greetForJoiningOrganization = async ({
-  user_id,
-  organization_id,
-}: {
-  user_id: number;
-  organization_id: number;
-}) => {
-  const userOrganisations = await getOrganizationsByUserId(user_id);
-  const organization = userOrganisations.find(
-    ({ id }) => id === organization_id,
-  );
-
-  if (isEmpty(organization)) {
-    throw new NotFoundError();
-  }
-
-  const { given_name, family_name, email } = (await findUserById(user_id))!;
-  const { cached_libelle, is_external, authentication_by_peers_type } =
-    organization;
-
-  // Welcome the user when he joins is first organization as he may now be able to connect
-  await sendMail({
-    to: [email],
-    subject: "Votre compte MonComptePro a bien été créé",
-    template: "welcome",
-    params: { given_name, family_name, email },
-  });
```

Comment:
delete the file but keep the mail!

Reply:
I can't move the call to the sendMail function outside the `join.ts` function because it will never be called, so I put it at the end of the `joinOrganization` function. Tell me if I'm wrong.